tests/test_runtime.py

```diff
@@ -128,6 +131,31 @@ async def error(*args, **kwargs):
 
         assert self.runtime.state == RuntimeStates.error, self.runtime.state
 
+    def test_cancelled_during_start_agent_loop(self, caplog):
+        """Test asyncio.CancelledError during _start_agent_loop."""
+
+        # we check for this log because async_utils.Runnable._thread_target
+        # catches all BaseExceptions and logs them instead of raising
```

Comment:
this one was a bit nasty to write, got some useful pair programming assistance from @0xArdi and left a comment for the reader to explain why `pytest.raises` cannot be used to catch the `asycio.CancelledError`